diff --git a/Makefile.libretro b/Makefile.libretro
index 6bacb3f..134f92e 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -599,6 +599,24 @@ else ifeq ($(platform), miyoo)
 	DYNAREC = ari64
 	HAVE_NEON = 0
 
+# MIYOOMINI
+else ifeq ($(platform), miyoomini)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	CFLAGS += -marm $(MYARCH) -fomit-frame-pointer
+	CFLAGS += -ffast-math -fdata-sections -ffunction-sections -fsingle-precision-constant -flto -fPIC
+	LDFLAGS += -flto -fPIC
+	CPU_ARCH := arm
+	OPTIMIZE := -Ofast -DNDEBUG=1
+	HAVE_NEON = 1
+	HAVE_NEON_ASM = 1
+	BUILTIN_GPU = neon
+	ARCH = arm
+	DYNAREC = ari64
+
+
 # Emscripten
 else ifeq ($(platform), emscripten)
    TARGET  := $(TARGET_NAME)_libretro_$(platform).bc
