diff --git a/Makefile.miyoo b/Makefile.miyoo
index ccecab6..bc3d9e4 100644
--- a/Makefile.miyoo
+++ b/Makefile.miyoo
@@ -3,7 +3,7 @@
 #########################
 
 # Default toolchain directory
-TOOLCHAIN_DIR=/opt/miyoo
+TOOLCHAIN_DIR=/opt/miyoomini-toolchain
 
 # All toolchain-related variables may be
 # overridden via the command line
@@ -26,11 +26,11 @@ else
 STRIP                 = $(TOOLCHAIN_DIR)/usr/bin/arm-linux-strip
 endif
 
-GCW0_SDL_CONFIG      ?= $(TOOLCHAIN_DIR)/usr/arm-miyoo-linux-uclibcgnueabi/sysroot/usr/bin/sdl-config
-GCW0_FREETYPE_CONFIG ?= $(TOOLCHAIN_DIR)/usr/arm-miyoo-linux-uclibcgnueabi/sysroot/usr/bin/freetype-config
+GCW0_SDL_CONFIG      ?= $(TOOLCHAIN_DIR)/arm-linux-gnueabihf/sysroot/usr/bin/sdl-config
+GCW0_FREETYPE_CONFIG ?= $(TOOLCHAIN_DIR)/usr/arm-linux-gnueabihf/sysroot/usr/bin/freetype-config
 
-GCW0_INC_DIR         ?= $(TOOLCHAIN_DIR)/usr/arm-miyoo-linux-uclibcgnueabi/sysroot/usr/include
-GCW0_LIB_DIR         ?= $(TOOLCHAIN_DIR)/usr/arm-miyoo-linux-uclibcgnueabi/sysroot/usr/lib
+GCW0_INC_DIR         ?= $(TOOLCHAIN_DIR)/arm-linux-gnueabihf/sysroot/usr/include
+GCW0_LIB_DIR         ?= $(TOOLCHAIN_DIR)/arm-linux-gnueabihf/sysroot/usr/lib
 
 #########################
 #########################
@@ -118,7 +118,8 @@ TARGET = retroarch
 
 OBJ :=
 LINK := $(CXX)
-DEF_FLAGS := -march=armv5te -mtune=arm926ej-s -ffast-math -fomit-frame-pointer
+DEF_FLAGS := -marm -mtune=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard -march=armv7ve
+DEF_FLAGS += -ffast-math -fomit-frame-pointer
 DEF_FLAGS += -ffunction-sections -fdata-sections
 DEF_FLAGS += -I. -Ideps -Ideps/stb -DMIYOO=1 -DDINGUX -MMD
 DEF_FLAGS += -Wall -Wno-unused-variable -flto
@@ -127,7 +128,7 @@ LIBS := -ldl -lz -lrt -pthread -lasound
 CFLAGS :=
 CXXFLAGS := -fno-exceptions -fno-rtti -std=c++11 -D__STDC_CONSTANT_MACROS
 ASFLAGS :=
-LDFLAGS := -Wl,--gc-sections
+LDFLAGS := -s =fast -Wl,--gc-sections
 INCLUDE_DIRS = -I$(GCW0_INC_DIR)
 LIBRARY_DIRS = -L$(GCW0_LIB_DIR)
 DEFINES := -DRARCH_INTERNAL -D_FILE_OFFSET_BITS=64 -UHAVE_STATIC_DUMMY
diff --git a/input/drivers_joypad/sdl_miyoomini_joypad.c b/input/drivers_joypad/sdl_miyoomini_joypad.c
index 0528946..651a404 100644
--- a/input/drivers_joypad/sdl_miyoomini_joypad.c
+++ b/input/drivers_joypad/sdl_miyoomini_joypad.c
@@ -38,9 +38,9 @@
 /* Vibration time at 100% vibration strength */
 #define SDL_MIYOOMINI_RUMBLE_MS   200
 
-/* Uncomment if you want the MENU button to be used exclusively for MENU
+// Uncomment if you want the MENU button to be used exclusively for MENU
 #define SDL_MIYOOMINI_HAS_MENU_TOGGLE
-*/
+//
 /* Miyoomini input map to keyboard keys */
 #define SDL_MIYOOMINI_SDLK_X      SDLK_LSHIFT
 #define SDL_MIYOOMINI_SDLK_A      SDLK_SPACE
